### PR TITLE
Add more ContentApp cluster handlers

### PR DESCRIPTION
**Pull request by chrisdecenzo**
#### Problem
* ContentApp clusters do not route messages to the ContentApp

#### Change overview
* Add cluster handlers for all ContentApp clusters

#### Testing
* chip-tool commands
* TODO: integration tests

// launch tv-app
./out/host/chip-tv-app --secured-device-port 5640 --secured-commissioner-port 5552

// commission it from chip-tool
./out/host/chip-tool pairing onnetwork-long 1234 20202021 3840

// launch app named "App3" which is a dummy app on linux, which puts it at endpoint 6
./out/host/chip-tool applicationlauncher launch-app foo1 1 App3 1234 1

// read the application launcher list
./out/host/chip-tool applicationlauncher read application-launcher-list 1234 6

// read the accepts header for content launcher
./out/host/chip-tool contentlauncher read accepts-header-list 1234 6

// use content launcher to launch a url
./out/host/chip-tool contentlauncher launch-url http://www.yahoo.com foo2 1234 6

// use content launcher to launch content
./out/host/chip-tool contentlauncher launch-content 1 data1 1234 6

// use target navigator to read list of targets
./out/host/chip-tool targetnavigator read target-navigator-list 1234 6

// use target navigator to navigate to target 2
./out/host/chip-tool targetnavigator navigate-target 2 foo1 1234 6

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Review by lazarkov — APPROVED** [review]

File: examples/tv-app/android/include/application-launcher/ApplicationLauncherManager.cpp [hunk 1]
```diff
@@ -56,3 +57,27 @@ ApplicationLauncherResponse applicationLauncherClusterLaunchApp(ApplicationLaunc
...
 }
+
+ApplicationLauncherResponse applicationLauncherClusterStopApp(chip::EndpointId endpoint, ApplicationLauncherApp application,
+                                                              std::string data)
+{
+    ChipLogProgress(Zcl, "ApplicationLauncherManager::applicationLauncherClusterStopApp");
+
+    ApplicationLauncherResponse response;
+    const char * testData = "data";
+    response.data         = (uint8_t *) testData;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/android/include/application-launcher/ApplicationLauncherManager.cpp:68]: fixed/no longer valid

File: examples/tv-app/android/include/application-launcher/ApplicationLauncherManager.cpp [hunk 2]
```diff
@@ -56,3 +57,27 @@ ApplicationLauncherResponse applicationLauncherClusterLaunchApp(ApplicationLaunc
...
+}
+
+ApplicationLauncherResponse applicationLauncherClusterHideApp(chip::EndpointId endpoint, ApplicationLauncherApp application,
+                                                              std::string data)
+{
+    ChipLogProgress(Zcl, "ApplicationLauncherManager::applicationLauncherClusterHideApp");
+
+    ApplicationLauncherResponse response;
+    const char * testData = "data";
+    response.data         = (uint8_t *) testData;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/android/include/application-launcher/ApplicationLauncherManager.cpp:80]: fixed/no longer valid

File: examples/tv-app/linux/AppImpl.cpp [hunk 3]
```diff
@@ -191,6 +191,34 @@ uint32_t AccountLoginImpl::GetSetupPIN(const char * tempAccountId)
...
+{
+    std::string appId(application.applicationId.data(), application.applicationId.size());
+    ChipLogProgress(DeviceLayer,
+                    "ApplicationLauncherResponse: LaunchApp application.catalogVendorId=%d "
+                    "application.applicationId=%s data=%s",
+                    application.catalogVendorId, appId.c_str(), data.c_str());
+
+    ApplicationLauncherResponse response;
+    const char * testData = "data";
+    response.data         = (uint8_t *) testData;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/AppImpl.cpp:204]: fixed/no longer valid

File: examples/tv-app/linux/AppImpl.cpp [hunk 4]
```diff
@@ -199,12 +227,12 @@ ContentAppFactoryImpl::ContentAppFactoryImpl()
 
 ContentApp * ContentAppFactoryImpl::LoadContentAppByVendorId(uint16_t vendorId)
 {
-    for (unsigned int i = 0; i < sizeof(mContentApps); i++)
+    for (unsigned int i = 0; i < APP_LIBRARY_SIZE; i++)
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/AppImpl.cpp:230]: fixed/no longer valid

File: examples/tv-app/linux/AppImpl.cpp [hunk 5]
```diff
@@ -213,6 +241,29 @@ ContentApp * ContentAppFactoryImpl::LoadContentAppByVendorId(uint16_t vendorId)
...
+{
+    std::string appId(application.applicationId.data(), application.applicationId.size());
+    ChipLogProgress(DeviceLayer,
+                    "ContentAppFactoryImpl: LoadContentAppByAppId application.catalogVendorId=%d "
+                    "application.applicationIdSize=%ld application.applicationId=%s ",
+                    application.catalogVendorId, application.applicationId.size(), appId.c_str());
+
+    for (unsigned int i = 0; i < APP_LIBRARY_SIZE; i++)
+    {
+        ContentAppImpl app = mContentApps[i];
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/AppImpl.cpp:254]: fixed

File: examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp [hunk 6]
```diff
@@ -34,25 +36,72 @@ CHIP_ERROR ApplicationLauncherManager::Init()
     return err;
 }
 
-CHIP_ERROR ApplicationLauncherManager::proxyGetApplicationList(chip::app::AttributeValueEncoder & aEncoder)
+CHIP_ERROR ApplicationLauncherManager::proxyGetApplicationList(chip::EndpointId mEndpointId,
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp:39]: fixed/no longer valid

File: examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp [hunk 7]
```diff
@@ -34,25 +36,72 @@ CHIP_ERROR ApplicationLauncherManager::Init()
...
         ReturnErrorOnFailure(encoder.Encode(456u));
         return CHIP_NO_ERROR;
     });
 }
 
-ApplicationLauncherResponse applicationLauncherClusterLaunchApp(ApplicationLauncherApp application, std::string data)
+ApplicationLauncherResponse applicationLauncherClusterLaunchApp(chip::EndpointId endpoint, ApplicationLauncherApp application,
+                                                                std::string data)
 {
+    ChipLogProgress(Zcl, "ApplicationLauncherManager::applicationLauncherClusterLaunchApp endpoint=%d", emberAfCurrentEndpoint());
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp:53]: fixed/no longer valid

File: examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp [hunk 8]
```diff
@@ -34,25 +36,72 @@ CHIP_ERROR ApplicationLauncherManager::Init()
...
 }
+
+ApplicationLauncherResponse applicationLauncherClusterStopApp(chip::EndpointId endpoint, ApplicationLauncherApp application,
+                                                              std::string data)
+{
+    ChipLogProgress(Zcl, "ApplicationLauncherManager::applicationLauncherClusterStopApp");
+
+    ApplicationLauncherResponse response;
+    const char * testData = "data";
+    response.data         = (uint8_t *) testData;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp:92]: fixed/no longer valid

File: examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp [hunk 9]
```diff
@@ -34,25 +36,72 @@ CHIP_ERROR ApplicationLauncherManager::Init()
...
+}
+
+ApplicationLauncherResponse applicationLauncherClusterHideApp(chip::EndpointId endpoint, ApplicationLauncherApp application,
+                                                              std::string data)
+{
+    ChipLogProgress(Zcl, "ApplicationLauncherManager::applicationLauncherClusterHideApp");
+
+    ApplicationLauncherResponse response;
+    const char * testData = "data";
+    response.data         = (uint8_t *) testData;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp:104]: fixed/no longer valid

File: src/app/clusters/target-navigator-server/target-navigator-server.cpp [hunk 10]
```diff
@@ -61,7 +61,7 @@ bool emberAfTargetNavigatorClusterNavigateTargetCallback(app::CommandHandler * c
 
     // TODO: char is not null terminated, verify this code once #7963 gets merged.
     std::string dataString(data.data(), data.size());
-    TargetNavigatorResponse response = targetNavigatorClusterNavigateTarget(target, dataString);
+    TargetNavigatorResponse response = targetNavigatorClusterNavigateTarget(emberAfCurrentEndpoint(), target, dataString);
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/clusters/target-navigator-server/target-navigator-server.cpp:64]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 11]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
     return EMBER_ZCL_STATUS_FAILURE;
 }
 
+EmberAfStatus KeypadInput::HandleReadAttribute(chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "KeypadInput::HandleReadAttribute: attrId=%d, maxReadLength=%d",
+                    static_cast<uint16_t>(attributeId), maxReadLength);
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:166]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 12]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus KeypadInput::HandleWriteAttribute(chip::AttributeId attributeId, uint8_t * buffer)
+{
+    ChipLogProgress(DeviceLayer, "KeypadInput::HandleWriteAttribute: attrId=%d", static_cast<uint16_t>(attributeId));
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:172]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 13]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus ApplicationLauncher::HandleReadAttribute(chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationLauncher::HandleReadAttribute: attrId=%d, maxReadLength=%d",
+                    static_cast<uint16_t>(attributeId), maxReadLength);
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:179]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 14]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus ApplicationLauncher::HandleWriteAttribute(chip::AttributeId attributeId, uint8_t * buffer)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationLauncher::HandleWriteAttribute: attrId=%d", static_cast<uint16_t>(attributeId));
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:185]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 15]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+    mCurrentTarget = currentTarget;
+}
+
+CHIP_ERROR TargetNavigator::GetTargetInfoList(chip::app::AttributeValueEncoder & aEncoder)
+{
+    ChipLogProgress(DeviceLayer, "TargetNavigator: GetTargetInfoList ");
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        int i = 0;
+        for (std::string entry : mTargets)
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:227]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 16]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+        {
+            // ReturnErrorOnFailure(encoder.Encode(chip::CharSpan(entry.c_str(), entry.length())));
+
+            chip::app::Clusters::TargetNavigator::Structs::NavigateTargetTargetInfo::Type targetInfo;
+            targetInfo.name       = chip::CharSpan(entry.c_str(), entry.length());
+            targetInfo.identifier = static_cast<uint8_t>(i++);
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:233]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 17]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+    });
+}
+
+TargetNavigatorResponse TargetNavigator::NavigateTarget(uint8_t target, std::string data)
+{
+    ChipLogProgress(DeviceLayer, "TargetNavigator: NavigateTarget target=%d data=\"%s\"", target, data.c_str());
+
+    TargetNavigatorResponse response;
+    const char * testData = "data response";
+    response.data         = (uint8_t *) testData;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:246]: fixed/no longer valid

File: src/app/util/ContentApp.cpp [hunk 18]
```diff
@@ -160,5 +160,137 @@ EmberAfStatus AccountLogin::HandleWriteAttribute(chip::AttributeId attributeId,
...
+}
+
+EmberAfStatus TargetNavigator::HandleReadAttribute(chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "TargetNavigator::HandleReadAttribute: attrId=%d, maxReadLength=%d",
+                    static_cast<uint16_t>(attributeId), maxReadLength);
+
+    if ((attributeId == ZCL_TARGET_NAVIGATOR_CURRENT_TARGET_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *(uint8_t *) buffer = mCurrentTarget;
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply src/app/util/ContentApp.cpp:266]: fixed/no longer valid

File: examples/tv-app/linux/AppImpl.cpp [hunk 19]
```diff
@@ -213,6 +241,29 @@ ContentApp * ContentAppFactoryImpl::LoadContentAppByVendorId(uint16_t vendorId)
...
 
+ContentApp * ContentAppFactoryImpl::LoadContentAppByAppId(ApplicationLauncherApp application)
+{
+    std::string appId(application.applicationId.data(), application.applicationId.size());
+    ChipLogProgress(DeviceLayer,
+                    "ContentAppFactoryImpl: LoadContentAppByAppId application.catalogVendorId=%d "
+                    "application.applicationIdSize=%ld application.applicationId=%s ",
+                    application.catalogVendorId, application.applicationId.size(), appId.c_str());
+
+    for (unsigned int i = 0; i < APP_LIBRARY_SIZE; i++)
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/linux/AppImpl.cpp:252]: changed to ArraySize(mContentApps)

**Pull request merged**